docs/searching-algorithms/linear-search.mdx

```diff
@@ -0,0 +1,175 @@
+---
+title: Linear Search
+slug: /searching-algorithms/linear-search
+---
+
+---
+title: Linear Search
+slug: /searching-algorithms/linear-search
+---
+
+# Introduction
+
+Linear search is the simplest of the search algorithms. The motive of any searching algorithm is to search for the presence of a key element in the collection of elements( arrays, linked lists, etc). Linear search starts the search of the key element(the element whose presence in the given list is to be searched) sequentially, from the beginning to the list until it is found. 
+
+# Algorithm
+
+In linear search, we follow the following algorithm:
+
+* Starting from the leftmost element of the array, compare each element with the key element
+
+* If the element compared is same as the key, then return the index of the element which matched the key
+
+* If the key does not match any element scanned in the array, then return -1
+
+
+# Explanation for Linear Search
+
+### Example 1
+
+```
+ Given the below array arr[4]:        
+ | 1 | 5 | 7 |
+
+ The key element = 5
+
+ The key is compared to the first element arr[0]=1
+ The key does not match  arr[0]
+
+ Next key is compared to arr[1] = 5
+ Key matches arr[2] and index 1 is returned
+
+```
+### Example 2
+
+```
+ Given the below array arr[2]:        
+ | 1 | 3 | 
+
+ The key element = 9
+
+ The key is compared to the first element arr[0]=1
+ The key does not match  arr[0]
+
+ Next key is compared to arr[1] = 3
+ Key = 5 does not match the arr[1]
+
+ The end of the array is reached. 
+ The key is not present in the array
+ -1 is returned indicating the absence of the key in array 
+
+```
```

Comment:
Use a smaller example that is easy to consume. Be minimal :)

Reply:
I have tried to use a simpler example with a smaller explanation